server/controller/program.controller.js

```diff
@@ -92,22 +92,25 @@ programController.get(
   async (req, res) => {
     const errors = validationResult(req);
     if (errors.isEmpty() === false) {
-      return res.status(400).json(errors);
+      generateServerErrorCode(res, 400, errors);
     }
     if (
       !req.query.id &&
       !req.query.school &&
       !req.query.major &&
       !req.query.catalogYear
     ) {
-      return res
-        .status(400)
-        .json({ error: ID_OR_ANY_OF_THREE_PARAMETERS_IS_REQUIRED });
+      generateServerErrorCode(
+        res,
+        400,
+        e,
+        ID_OR_ANY_OF_THREE_PARAMETERS_IS_REQUIRED
+      );
     }
 
-    const formattedQuery = {};
+    let formattedQuery = {};
```

Comment:
Does formattedQuery change later?

Reply:
Yes. Properties are added to the object later on inside the following forEach callback function.

Reply:
when formattedQuery change means you assign it to some other object.
`formattedQuery = otherObject;`
 But if you're adding key-value to the Object, formattedQuery remain unchanged

Reply:
Oh. I didn't know that's how const on objects work. Thanks.